editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,352 @@
+      return;
+    } else if (name.getText().length() > 62) {
+      new Alert(Alert.AlertType.ERROR, "Version name must not exceed 62 characters").show();
+      return;
+    } else if (name.getText().length() == 0) {
+      new Alert(Alert.AlertType.ERROR, "Please enter a version name").show();
+      return;
+    }
+
+    // validate version access input
+    if (accessComboBox.getSelectionModel().getSelectedItem() == null) {
+      new Alert(Alert.AlertType.ERROR, "Please select an access level").show();
+      return;
+    }
+
+    // set the user defined name, access level and description as service version parameters
+    ServiceVersionParameters newVersionParameters = new ServiceVersionParameters();
+    newVersionParameters.setName(name.getText());
+    newVersionParameters.setAccess(accessComboBox.getSelectionModel().getSelectedItem());
+    newVersionParameters.setDescription(description.getText());
+
+    // update the UI
+    createVersionButton.setText("Creating version....");
+    createVersionButton.setDisable(true);
+
+    // create a new version with the specified parameters
+    ListenableFuture<ServiceVersionInfo> createdVersion = serviceGeodatabase.createVersionAsync(newVersionParameters);
+    createdVersion.addDoneListener(() -> {
+      try {
+        // get the name of the created version and switch to it
+        ServiceVersionInfo createdVersionInfo = createdVersion.get();
+        userCreatedVersion = createdVersionInfo.getName();
+        switchVersion(userCreatedVersion);
+
+        // hide the form from the UI as the sample only allows 1 version to be created
+        createVersionVBox.setVisible(false);
+        switchVersionButton.setDisable(false);
+      } catch (Exception ex) {
+        // if there is an error creating a new version, display an alert and reset the UI
+        if (ex.getCause().toString().contains("The version already exists")) {
+          new Alert(Alert.AlertType.ERROR, "Please enter a unique version name").show();
+        } else {
+          new Alert(Alert.AlertType.ERROR, "Error creating new version").show();
+        }
+        createVersionButton.setText("Create version");
+        createVersionButton.setDisable(false);
+      }
+    });
+  }
+
+  /**
+   * Apply local edits to the service geodatabase and switch branch version
+   */
+  @FXML
+  private void handleSwitchVersionButtonClicked() {
+    if (serviceGeodatabase.getVersionName().equals(userCreatedVersion)) {
+      // if the user created version has local edits, apply the edits to the service geodatabase
+      if (serviceGeodatabase.hasLocalEdits()) {
+        ListenableFuture<List<FeatureTableEditResult>> resultOfApplyEdits = serviceGeodatabase.applyEditsAsync();
+        resultOfApplyEdits.addDoneListener(() -> {
+          try {
+            // check if the server edit was successful
+            List<FeatureTableEditResult> edits = resultOfApplyEdits.get();
+            if (edits == null || edits.size() == 0) {
```

Comment:
A better way to check if a collection is empty is `.isEmpty()`:
```suggestion
            if (edits == null || edits.isEmpty()) {
```